python/cli/prelude_cli/views/build.py

```diff
@@ -149,25 +149,35 @@ def upload(p: Path):
 @click.option('-s', '--source', help='source of threat (ex. www.cisa.gov)', default=None, type=str)
 @click.option('-i', '--source_id', help='ID of the threat, per the source (ex. aa23-075a)', default=None, type=str)
 @click.option('-t', '--tests', help='comma-separated list of test IDs', default=None, type=str)
-@click.option('-d', '--directory', help='directory containing tests and IOAs generated from threat_intel', default=None, type=click.Path(exists=True, dir_okay=True, file_okay=False))
+@click.option('-d', '--directory', help='directory containing tests, detections, and hunt queries generated from threat_intel', default=None, type=click.Path(exists=True, dir_okay=True, file_okay=False))
 @click.pass_obj
 @handle_api_error
 def create_threat(controller, name, published, id, source_id, source, tests, directory):
     """ Create a security threat """
     with Spinner(description='Creating new threat'):
         try:
-            created_tests = list()
-            uploads = list()
+            created_tests = []
+            test_uploads = []
+            created_detections = []
+            created_queries = []
             threat = None
             if directory:
                 for technique_dir in os.listdir(directory):
-                    with open(f'{directory}/{technique_dir}/test.go', 'r') as f:
-                        go_code = f.read()
                     with open(f'{directory}/{technique_dir}/config.json', 'r') as f:
                         config = json.load(f)
-                    test = controller.create_test(name=config['name'], unit=config['unit'], technique=config['technique'])
-                    created_tests.append(test)
-                    uploads.append(controller.upload(test_id=test['id'], filename=f'{test["id"]}.go', data=go_code.encode()))
+                        test = controller.create_test(name=config['name'], unit=config['unit'], technique=config['technique'])
+                        created_tests.append(test)
+                    with open(f'{directory}/{technique_dir}/test.go', 'r') as f:
+                        go_code = f.read()
+                        test_uploads.append(controller.upload(test_id=test['id'], filename=f'{test["id"]}.go', data=go_code.encode()))
+                    for sigma_file in Path(f'{directory}/{technique_dir}').glob('sigma*'):
+                        with open(sigma_file, 'r') as f:
+                            rule = f.read()
+                            created_detections.append(controller.create_detection(rule=rule, test_id=test['id']))
+                    for query_file in Path(f'{directory}/{technique_dir}').glob('query*'):
+                        with open(query_file, 'r') as f:
+                            queries = f.read()
+                            created_queries.append(controller.create_threat_hunt(name=config['name'], query=queries, test_id=test['id']))
```

Comment:
each query should have a specific name- not the test name

Reply:
done